sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -530,6 +530,88 @@ def estimate_size(self, unused_value, nested=False):
     return 1
 
 
+class MapCoderImpl(StreamCoderImpl):
+  """For internal use only; no backwards-compatibility guarantees.
+
+  A coder for typing.Mapping objects."""
+  def __init__(
+      self,
+      key_coder,  # type: CoderImpl
+      value_coder  # type: CoderImpl
+  ):
+    self._key_coder = key_coder
+    self._value_coder = value_coder
+
+  def encode_to_stream(self, value, out, nested):
+    size = len(value)
+    out.write_bigendian_int32(size)
```

Comment:
We don't care about larger maps? Also, why not varint? (If that's what's conventionally used elsewhere, I'm fine with that, but iterable coder uses varint64.)

Reply:
Yeah I thought about that too. I was just trying to exactly replicate what we do in [MapCoder.java] (https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/MapCoder.java) for now though. Do you think I should switch this and MapCoder.java over to using varint here?

Reply:
This is unfortunate. We can't change MapCoder because that would be backwards incompatible (and users might have encoded Maps stored in states or elsewhere), so I suppose we'll have to stick with this limitation.

Reply:
As a practical matter, int32 max of any amount of data is bigger than our GRPC limits for receiving single values into an SDK. At least, until we add a large value protocol to stream in single large values somewhere.